### DISCUSSION TITLE
Multiple Gensets?

### DISCUSSION THREAD

**Discussion by Twilliamson3** [jgyates/genmon#1193] [General]
Could 1 instance of this software monitor multiple generators. I have about 15 all on the same ip network that i would like to connect into genmon. Can this be done? I supposed i could run a separate docker instance for each one, but that's less than ideal.

**Comment by jgyates** [comment]
This wiki page has info in running multiple instances of genmon on one system. Monitoring multiple generators with one instance is not supported.

https://github.com/jgyates/genmon/wiki/Appendix-J-Multiple-Instances-of-Genmon

**Comment by skipfire** [comment]
What is your goal?  Are you wanting to minimize dockers or try to get a single consolidated view of all your generators at once?

**Comment by Twilliamson3** [comment]
Trying to get a single view of all my assets at once. Ideally be able to
SNMP them into Zabbix for data collection / dashboards.

**Reply by jgyates** [reply]
I don't know that multiple instances will work with SNMP. I have not tested this add on with multiple instances. The genmon add on for SNMP uses the python library pysnmp to add support for SNMP and I do not know if that will work with multiple instances on a single IP address.  At a minimum you would need to set each SNMP add on instances with a unique Enterprise ID in the add on settings, but if it will work depends on the pysnmp library design implementation. SNMP requires a socket listener on an IP address (genmon) and  that is not shared between the instances so I am guessing SNMP will not work. The MQTT add on should work with no issues as each instances output to an exernal server socket on a broker IP address. 

If you decide to try this out I can help with questions.
